confidence/models.py

```diff
@@ -233,36 +229,17 @@ def __setstate__(self, state: typing.Dict[str, typing.Any]) -> None:
 
         if isinstance(self._missing, Missing):
             # reverse the Missing encoding done in __getstate__
-            self._missing = _MISSING_MAPPING[self._missing]
-
-
-class _NotConfigured(Configuration):
-    """
-    Sentinel value to signal there is no value for a requested key.
-    """
-    def __bool__(self) -> bool:
-        return False
-
-    def __repr__(self) -> str:
-        return '(not configured)'
-
-    __str__ = __repr__
+            self._missing = {Missing.silent: NotConfigured,
+                             Missing.error: _NoDefault}[self._missing]
 
 
-# TODO: can't reference NotConfigured yet here, but instantiation needs the value, maybe a cleaner solution is needed?
-_MISSING_MAPPING: typing.MutableMapping[Missing, typing.Any] = {Missing.silent: None,
-                                                                Missing.error: NoDefault}
-
-
-# FIXME: ordering of things in this file is a mess (some of it because of reasons, needs cleanup)
-
-
-# overwrite NotConfigured as an instance of itself, a Configuration instance without any values
-NotConfigured = _NotConfigured()
-# NB: NotConfigured._missing refers to the NotConfigured *class* at this point, fix this after the name override
+NotConfigured = type("NotConfigured", (Configuration,), {
+    "__bool__": lambda self: False,
+    "__repr__": lambda self: '(not configured)',
+    "__str__": lambda self: '(not configured)',
```

Comment:
Would a `__doc__` member here be picked up as the docstring?

Reply:
I think it would. The following passes:

```python
    import inspect
    A = type("A", (object,), {"__doc__": "(docstring)"})
    assert inspect.getdoc(A()) == "(docstring)"
```

Reply:
Added a the old docstring for `NotConfigured` back through the `__doc__` member